draft-ietf-quic-tls.md

```diff
@@ -385,13 +385,9 @@ perspective of the endpoint in question.
 
 ### Handshake Confirmed {#handshake-confirmed}
 
-In this document, the TLS handshake is considered confirmed at an endpoint when
-the following two conditions are met: the handshake is complete, and the
-endpoint has received an acknowledgment for a packet sent with 1-RTT keys.
-This second condition can be implemented by recording the lowest packet number
-sent with 1-RTT keys, and the highest value of the Largest Acknowledged field
-in any received 1-RTT ACK frame: once the latter is higher than or equal to the
-former, the handshake is confirmed.
+In this document, the TLS handshake is considered confirmed at the server when
+the handshake completes. At the client, the handshake is considered confirmed
+when the HANDSHAKE_DONE frame is received.
 
```

Comment:
Is it clear in which PN space the HANDSHAKE_DONE frame is sent in? To me it seems it could be in either 1-RTT or HS PN. But going forward - since HS keys must be dropped and done must be transmitted, it then follows it must be 1-RTT.

Reply:
Handshake is done at the server upon receipt of the ClientFinished.  Once it's done, the frame is transmitted by the server, and the server is using 1-RTT keys to send.  Therefore, it has to be 1-RTT.

However, I can no longer find the section specifying which frame types are allowed in which encryption level.  This should probably be listed there, and if we've lost that text, we might have bigger problems.

Reply:
https://quicwg.org/base-drafts/draft-ietf-quic-tls.html#rfc.section.4 says
> All other frame types MUST only be sent in the 0-RTT and 1-RTT levels.

And since the frame can't be sent by the client, this means that the frame can only be sent in 1-RTT packets.